### PR TITLE
add provider info

### PR DESCRIPTION
**Pull request by r10s**
sth. as https://github.com/deltachat/deltachat-android/pull/1219 on android and https://github.com/deltachat/deltachat-desktop/pull/1416 on desktop

todo:

- show PREPARATION in yellow and BROKEN in red, maybe inside the footer or using a separate cell.

- add a "more info" link/button, the link should be not too big, should not be mixed up by the user with the login button, sth. as the android/desktop screenshots show (links above) should be fine. the button will open the browser with the url from provider.overviewPage

- the provider info should be **hidden**, once a different or no provider is returned from the email-address - in contrast, it should only be **shown** on focus changes or login-tap (this is because of the upcoming oauth2 stuff)

### CODE REVIEW

**Comment by nayooti** [comment]
what exactly should the 'info button' do?

**Comment by nayooti** [comment]
@r10s I finished it except for the functionality of the info button.

**Comment by r10s** [comment]
@nayooti the info button shall open the external browser with the url provided by DcProvider. getOverviewPage

**Comment by nayooti** [comment]
@r10s It's all implemented. Could not assign you to review.

File: deltachat-ios/View/ProviderInfoCell.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    private let fontSize: CGFloat = 13
+
+    private let brokenColorSet: ColorSet = ColorSet(
+        backgroundColor: SystemColor.red.uiColor,
+        tintColor: UIColor.white
+    )
+
+    private let preparationColorSet: ColorSet = ColorSet(
```
Comment (r10s) [inline deltachat-ios/View/ProviderInfoCell.swift]: for preparation-color, can we please use the colors from https://github.com/deltachat/deltachat-android/pull/1219/files#diff-f189d9ec03b8e9f14e2941b3ad306427R62 - the red for broken looks good and it is probably better to use the system-alert-color for that,

File: deltachat-ios/View/ProviderInfoCell.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        hintBackgroundView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -margin).isActive = true
+
+        hintLabel.leadingAnchor.constraint(equalTo: hintBackgroundView.leadingAnchor, constant: padding).isActive = true
+        hintLabel.topAnchor.constraint(equalTo: hintBackgroundView.topAnchor, constant: 5).isActive = true
+        hintLabel.trailingAnchor.constraint(equalTo: hintBackgroundView.trailingAnchor, constant: -padding).isActive = true
+
+        infoButton.topAnchor.constraint(equalTo: hintLabel.bottomAnchor, constant: padding).isActive = true
+        infoButton.leadingAnchor.constraint(equalTo: hintBackgroundView.leadingAnchor, constant: padding).isActive = true
+        infoButton.bottomAnchor.constraint(equalTo: hintBackgroundView.bottomAnchor, constant: -5).isActive = true
+    }
```
Comment (r10s) [inline deltachat-ios/View/ProviderInfoCell.swift]: the padding atop of the text should be roughly doubled.

the font has to be a bit larger, maybe sth. in between the current size and the size used for the control labels.

File: deltachat-ios/Controller/AccountSetupController.swift [hunk 3]
```diff
@@ -101,15 +106,15 @@ class AccountSetupController: UITableViewController {
...
     private func updateProgressHudValue(value: Int?) {
         if let value = value {
             configProgressAlert.message = String.localized("one_moment") + " " + String(value/10) + "%"
         }
     }
-
-    // cells
-
+    
+    // MARK: -cells
```
[the remark this answers is not in the document]
Comment (nayooti) [reply deltachat-ios/Controller/AccountSetupController.swift]: Not sure, what's wrong with the MARK-comments. I believe, I am using exactly the format they are suggesting.

[the remark this answers is not in the document]
Comment (nayooti) [reply deltachat-ios/Controller/AccountSetupController.swift]: ok, found it: whitespace after '-'

[the remark this answers is not in the document]
Comment (r10s) [reply deltachat-ios/Controller/AccountSetupController.swift]: +1

File: deltachat-ios/Controller/AccountSetupController.swift [hunk 4]
```diff
@@ -493,112 +530,143 @@ class AccountSetupController: UITableViewController {
...
-
-         	})
-         	oAuthAlertController.addAction(confirm)
-         	oAuthAlertController.addAction(cancel)
-
-         	present(oAuthAlertController, animated: true, completion: nil)
-         	return true
+         let title = "Continue with simplified setup"
+         // swiftlint:disable all
+         let message = "The entered e-mail address supports a simplified setup (oAuth2).\n\nIn the next step, please allow Delta Chat to act as your Chat with E-Mail app.\n\nThere are no Delta Chat servers, your data stays on your device."
```
Comment (r10s) [inline deltachat-ios/Controller/AccountSetupController.swift]: also here  @nayooti your editor is doing strange things and reformats unaffected code. i am currently fixing that, just saying.

wrt. the empty lines, in xcode, you have to enable "Preferences / Text Editting / Editing / Including whitespace only lines"

**Comment by r10s** [comment]
@nayooti great! ready to merge, i'd say :)

**Pull request merged**